src/dolphin/atmosphere/_netcdf.py

```diff
@@ -0,0 +1,518 @@
+from __future__ import annotations
+
+import datetime
+import itertools
+import logging
+import multiprocessing as mp
+from collections import defaultdict
+from pathlib import Path
+from typing import Iterable, Optional, Union
+
+import numpy as np
+import xarray
+from pyproj import CRS, Transformer
+from scipy.interpolate import RegularGridInterpolator as Interpolator
+
+from dolphin._types import Filename, TropoModel
+
+logger = logging.getLogger(__name__)
+
+###########
+# Mostly inherited from RAiDER
+
+__all__ = ["delay_from_netcdf"]
+
+
+def delay_from_netcdf(
+    dt: datetime.datetime,
+    weather_model_file: Filename,
+    SNWE: tuple[float, float, float, float],
+    height_levels: Optional[list[float]],
+    out_proj: Union[int, str] = 4326,
+    weather_model: TropoModel = TropoModel.ECMWF,
+):
+    """Calculate integrated delays on query points.
+
+    Parameters
+    ----------
+    dt: Datetime
+        Datetime object for determining when to calculate delays
+    weather_model_file: string
+        Name of the NETCDF file containing a pre-processed weather model
+    SNWE: tuple[float]
+        Bounding box in lat/lon
+    height_levels: list
+        (optional) list of height levels on which to calculate delays.
+        Only needed for cube generation.
+    out_proj: int,str
+        (optional) EPSG code for output projection
+    weather_model: TropoModel
+        weather model  (ECMWF or HRES)
+
+    Returns
+    -------
+    xarray Dataset *or* ndarrays:
+        wet and hydrostatic delays at the grid nodes / query points.
+
+    """
+    # Load CRS from weather model file
+    with xarray.load_dataset(weather_model_file) as ds:
+
+        if "hydro" in ds:
+            wmodel_file = weather_model_file
+        else:
+            wmodel_file = convert_netcdf(
+                weather_model_file=weather_model_file,
+                lat_lon_bounds=SNWE,
+                weather_model_output_dir="./tmp",
+                weather_model=weather_model,
+            )
+
+        try:
+            wm_proj = CRS.from_wkt(ds["proj"].attrs["crs_wkt"])
+        except KeyError:
+            logger.warning(
+                "WARNING: I can't find a CRS in the weather model file, "
+                "so I will assume you are using WGS84"
+            )
+            wm_proj = CRS.from_epsg(4326)
+
+    # get heights
+    with xarray.load_dataset(wmodel_file) as ds:
+        wm_levels = ds.z.values
+
+    if height_levels is None:
+        height_levels = wm_levels
+
+    # TODO: expose this as library function
+    ds = _get_delays_on_cube(
+        dt, wmodel_file, wm_proj, SNWE, height_levels, out_epsg=out_proj
+    )
+
+    return ds
+
+
+def transform_bbox(snwe_in, dest_crs=4326, src_crs=4326, margin=100.0):
+    """Transform bbox to lat/lon or another CRS for use with rest of workflow.
+
+    Returns: SNWE.
+    """
+    # TODO - Handle dateline crossing
```

Comment:
Hmm.... does this mean it definitely will fail on the antimeridian? 
Let's note somewhere to check a workflow for this on the antimeridian crossing in Alaska.

[I have one way of converting multipolygons split across -180 in `burst_db`](https://github.com/opera-adt/burst_db/blob/main/src/burst_db/build_frame_db.py#L269). The general idea is warp to UTM (which could be only one of two zones) so there's not a coordinate jump, work with the coordinates, then warp back

Reply:
Thank you @scottstanie, yes let's check it separately later, I'll make an issue for that